pkg/clusteragent/autoscaling/externalmetrics/datadogmetric_controller.go

```diff
@@ -92,30 +101,37 @@ func NewDatadogMetricController(client dynamic.Interface, informer dynamicinform
 }
 
 // Run starts the controller to handle DatadogMetrics
-func (c *DatadogMetricController) Run(ctx context.Context) {
+func (c *DatadogMetricController) Run(ctx context.Context, numWorkers int) {
 	if ctx == nil {
 		log.Errorf("Cannot run with a nil context")
 		return
 	}
 	c.context = ctx
 
-	defer c.workqueue.ShutDown()
-
 	log.Infof("Starting DatadogMetric Controller (waiting for cache sync)")
 	if !cache.WaitForCacheSync(ctx.Done(), c.synced) {
 		log.Errorf("Failed to wait for DatadogMetric caches to sync")
 		return
 	}
 
-	go wait.Until(c.worker, time.Second, ctx.Done())
+	for i := 0; i < numWorkers; i++ {
+		go c.worker(i)
```

Comment:
should we pass the context ctx?

Reply:
The `workqueue` works as a channel and shutdown will be notified through it, so context is not necessary.